pom.xml

```diff
@@ -263,7 +269,7 @@
         </plugin>
         <plugin>
           <artifactId>maven-javadoc-plugin</artifactId>
-          <version>3.0.1</version>
+          <version>3.1.0</version>
```

Comment:
Could be taken from the parent pom

Reply:
That would mean delaying this pull request until https://github.com/jenkinsci/plugin-pom/pull/178 is merged and a new parent pom is released.  The parent pom currently provides maven-javadoc-plugin 3.0.1.  When I run a Java 11 build without this change, I get:

```
$ mvn clean -Djenkins.version=2.164 javadoc:javadoc
[INFO] --- maven-javadoc-plugin:3.0.1:javadoc (default-cli) @ git-client ---
[INFO]
2 errors
[ERROR] Error while creating javadoc report:
Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.

Command line was: /usr/lib/jvm/java-11-openjdk-amd64/bin/javadoc @options @packages

Refer to the generated Javadoc files in '/home/mwaite/git/jenkins/git-client-plugin/target/site/apidocs' dir.

org.apache.maven.reporting.MavenReportException:
Exit code: 1 - javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.
javadoc: error - The code being documented uses modules but the packages defined in https://docs.oracle.com/javase/8/docs/api/ are in the unnamed module.

Command line was: /usr/lib/jvm/java-11-openjdk-amd64/bin/javadoc @options @packages

Refer to the generated Javadoc files in '/home/mwaite/git/jenkins/git-client-plugin/target/site/apidocs' dir.

    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeJavadocCommandLine (AbstractJavadocMojo.java:5298)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.executeReport (AbstractJavadocMojo.java:2134)
    at org.apache.maven.plugins.javadoc.JavadocReport.generate (JavadocReport.java:134)
    at org.apache.maven.plugins.javadoc.JavadocReport.doExecute (JavadocReport.java:329)
    at org.apache.maven.plugins.javadoc.AbstractJavadocMojo.execute (AbstractJavadocMojo.java:1912)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

```

I'm happy to wait for a new version of the parent pom so that I can use the version from the parent pom instead.  Any idea when https://github.com/jenkinsci/plugin-pom/pull/178 will be merged and a new parent pom released?

Reply:
I made a mistake in the preceding comment.  I used the Java 11 binary delivered with Debian Testing (Debian 10 upcoming release).  It calls itself Java 11.0.3 and gives that stack trace.

If I use JDK 11.0.2 on that same machine (as installed from https://jdk.java.net/11/), it is JDK 11.0.2 and runs as expected.

I'll remove the setting of maven-javadoc-plugin version from this pull request.

Reply:
Done. Thanks!